src/main/java/uk/gov/hmcts/reform/sscs/service/SubmitAppealServiceBase.java

```diff
@@ -145,125 +140,153 @@ public Long submitAppeal(SyaCaseWrapper appeal, String userToken) {
         }
 
         log.info("{} - setting venue name to {}",
-                sscsCaseData.getAppeal().getAppellant().getIdentity().getNino(),
-                sscsCaseData.getProcessingVenue());
+            isIba
+                ? sscsCaseData.getAppeal().getAppellant().getIdentity().getIbcaReference()
+                : sscsCaseData.getAppeal().getAppellant().getIdentity().getNino(),
+            sscsCaseData.getProcessingVenue());
 
         return sscsCaseData;
     }
 
     private EventType findEventType(SscsCaseData caseData, boolean saveAndReturnCase) {
-
-        if (caseData.getAppeal().getMrnDetails() != null && caseData.getAppeal().getMrnDetails().getMrnDate() != null) {
-            LocalDate mrnDate = LocalDate.parse(caseData.getAppeal().getMrnDetails().getMrnDate());
+        Appeal appeal = caseData.getAppeal();
+        MrnDetails mrnDetails = appeal.getMrnDetails();
+        if (mrnDetails != null && mrnDetails.getMrnDate() != null) {
+            LocalDate mrnDate = LocalDate.parse(mrnDetails.getMrnDate());
             boolean moveToNoneCompliant = mrnDate.plusMonths(13L).isBefore(LocalDate.now());
 
             return handleMoveToNonCompliant(caseData, saveAndReturnCase, moveToNoneCompliant);
         } else {
-            log.info("Moving case for NINO {} to incomplete due to MRN Details {} present and MRN Date {} present",
-                    caseData.getAppeal().getAppellant().getIdentity().getNino(),
-                    (caseData.getAppeal().getMrnDetails() != null ? "" : "not"),
-                    (caseData.getAppeal().getMrnDetails().getMrnDate() != null ? "" : "not"));
+            boolean isIba = caseData.getBenefitCode().equals(Benefit.INFECTED_BLOOD_APPEAL.getBenefitCode());
+            Identity identity = appeal.getAppellant().getIdentity();
+            log.info("Moving case for {} {} to incomplete due to MRN Details {} present and MRN Date {} present",
+                isIba ? "Ibca Reference" : "NINO",
+                isIba ? identity.getIbcaReference() : identity.getNino(),
+                (mrnDetails != null ? "" : "not"),
+                (mrnDetails != null && mrnDetails.getMrnDate() != null ? "" : "not"));
             return saveAndReturnCase ? DRAFT_TO_INCOMPLETE_APPLICATION : INCOMPLETE_APPLICATION_RECEIVED;
         }
     }
 
-    private SscsCaseDetails createOrUpdateCase(SscsCaseData caseData, EventType eventType, IdamTokens idamTokens) {
+    private Pair<List<SscsCaseDetails>, SscsCaseDetails> getMatchedNinoCasesAndCaseDetails(SscsCaseData caseData,
+                                                                                           IdamTokens idamTokens,
+                                                                                           boolean isIba,
+                                                                                           String nino,
+                                                                                           String ibcaReference) {
```

Comment:
why pass isIba, Nino and ibcaReference as arguments if you're also passing caseData?

Reply:
these were defined in the function level above this so didn't see point in getting the values from the case data when we have them stored as variables anyway